Stripe.hs

```diff
@@ -0,0 +1,46 @@
+module Stripe
+    ( runStripe
+    , subscribeToPlan
+    , cancelSubscription
+    ) where
+
+import Import
+
+import qualified Web.Stripe as S
+import qualified Web.Stripe.Customer as S
+import qualified Web.Stripe.Subscription as S
+
+runStripe :: FromJSON a => S.Stripe a -> Handler (Maybe a)
+runStripe f = do
+    stripeKey <- stripeKeysSecretKey . appStripeKeys <$> getYesod
+    result <- liftIO $ S.stripe (S.StripeConfig $ encodeUtf8 stripeKey) f
+
+    case result of
+        Left err -> do
+            $(logError) $ pack $ show err
+            return Nothing
+        Right x -> return $ Just x
+
+subscribeToPlan :: S.TokenId
+                -> S.PlanId
+                -> Maybe S.CustomerId
+                -> S.Stripe S.CustomerId
+subscribeToPlan token planId mstripeId = do
+    stripeId <- findOrCreateCustomer token mstripeId
+
+    void $ S.createSubscription stripeId planId []
+
+    return stripeId
+
+cancelSubscription :: Maybe S.CustomerId -> S.Stripe ()
+cancelSubscription Nothing = return ()
+cancelSubscription (Just stripeId) = do
+    subs <- S.list <$>
+        S.getSubscriptions stripeId (Just 1) Nothing Nothing
```

Comment:
http://hackage.haskell.org/package/stripe-haskell-0.1.3.1/docs/Web-Stripe-ApplicationFee.html#t:Limit

It's the `Limit`. Just says return, at most `1` subscriptions. We expect users to only have 1 at any given time.